### PR TITLE
Add page with list of extensions

### PR DESCRIPTION
**Pull request by mateiidavid**
Addresses #1105 

### What

Something that I have noticed a few times when picking up questions on support is that we don't have a definitive list for Linkerd extensions. This PR addresses that by adding a new page with a list of extensions under `/linkerd.io/2.10/reference`.

To make it easier to add or remove extensions (and to encourage members of the community to add their own), the extensions are not hardcoded. Under `/data/` we now keep a yaml list for the extensions to be displayed on the page. Each list item has:
  * a name;
  * a description;
  * and a link to documentation/repository.

To start, I have added in the extensions that I know of, the built-in (default) extensions, linkerd-buoyant and tapshark.

Hopefully through this change we'll get to see more extensions pop up, or at the very least encourage the use of third party extensions.

### Rendered page

image

### CODE REVIEW

File: linkerd.io/content/2.10/reference/extension-list.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++
+
+Linkerd provides a mix of built-in and third-party
+[extensions](../../tasks/extensions/) to add additional functionality to the
+base installation. The following is the list of known extensions:
+
+{{< extensions-2-10 >}}
+
+If you have an extension for Linkerd and it is not on the list, [please submit
+a pull request!](https://github.com/linkerd/website)
```
Comment (alpeb) [inline linkerd.io/content/2.10/reference/extension-list.md:13]: May I suggest a more direct link to edit the list?
```suggestion
If you have an extension for Linkerd and it is not on the list, [please edit
this
page!](https://github.com/linkerd/website/edit/main/linkerd.io/data/extension-list.yaml)
```

**Review by alpeb — APPROVED** [review]
Awesome!
I think it's fine to keep the `extension-list.yaml` in this repo and not complicate ourselves with syncing it against the linkerd2 repo :+1: 
I added a suggestion to make its update easier :wink:

**Review by cpretzer — APPROVED** [review]
lgtm 🚢

**Comment by cpretzer** [comment]
I'll fix the lint errors in a different commit

**Pull request merged**